src/printer.js

```diff
@@ -50,26 +48,38 @@ const {
  * @param {(path: import('prettier').AstPath<any>) => import('prettier').Doc} print
  */
 function printTopLevelParts(node, path, opts, print) {
-  const parts = {
-    frontmatter: [],
-    markup: [],
-    styles: [],
-  };
+  let docs = [];
 
+  const normalize = (doc) => [stripTrailingHardline(doc), softline];
+
+  // frontmatter always comes first
   if (node.module) {
-    parts.frontmatter.push(path.call(print, 'module'));
+    const subDoc = normalize(path.call(print, 'module'));
+    docs.push(subDoc);
   }
 
-  if (node.css && node.css.length) {
-    parts.styles.push(path.call(print, 'css'));
+  // markup and styles follow, whichever the user prefers (default: markup, styles)
+  for (const section of parseSortOrder(opts.astroSortOrder)) {
+    switch (section) {
+      case 'markup': {
+        const subDoc = path.call(print, 'html');
+        if (!isEmptyDoc(subDoc)) docs.push(normalize(subDoc));
+        break;
+      }
+      case 'styles': {
+        const subDoc = path.call(print, 'css');
+        if (!isEmptyDoc(subDoc)) docs.push(normalize(subDoc));
+        break;
+      }
+    }
   }
 
-  if (node.html && !isObjEmpty(node.html)) {
-    parts.markup.push(path.call(print, 'html'));
-  }
+  // remove trailing softline, if any
+  const lastDoc = docs[docs.length - 1];
+  const lastItem = lastDoc[lastDoc.length - 1];
+  if (lastItem.type === 'line') docs[docs.length - 1].pop();
 
-  const docs = flatten([parts.frontmatter, ...parseSortOrder(opts.astroSortOrder).map((p) => parts[p])]).filter((doc) => '' !== doc);
```

Comment:
Could you re-add the implementation of allowing for respecting what `opts.astroSortOrder` is set to? It allows for choosing whether styles or HTML go first.

Reply:
Right that’s implemented in the `switch` statement! But that’s valid feedback—it needs a comment or something because it’s not very straightforward

Reply:
Oops I totally missed the switch statement first time around!